zkevm-circuits/src/evm_circuit/util/math_gadget/less.rs

```diff
@@ -0,0 +1,159 @@
+        // Set the bytes of diff
+        let diff = (lhs - rhs) + (if lt { self.range } else { F::zero() });
+        let diff_bytes = diff.to_repr();
+        for (idx, diff) in self.diff.iter().enumerate() {
+            diff.assign(
+                region,
+                offset,
+                Value::known(F::from(diff_bytes[idx] as u64)),
+            )?;
+        }
+
+        Ok((if lt { F::one() } else { F::zero() }, diff_bytes.to_vec()))
+    }
+
+    pub(crate) fn diff_bytes(&self) -> Vec<Cell<F>> {
+        self.diff.to_vec()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::super::test_util::*;
+    use super::*;
+    use eth_types::*;
+    use halo2_proofs::halo2curves::bn256::Fr;
+    use halo2_proofs::plonk::Error;
+
+    #[test]
+    fn test_lt() {
+        const N: usize = 3;
+        #[derive(Clone)]
+        /// a < b
+        struct LtGadgetTestContainer<F> {
+            lt_gadget: LtGadget<F, N>,
+            a: Cell<F>,
+            b: Cell<F>,
+        }
+
+        impl<F: Field> MathGadgetContainer<F> for LtGadgetTestContainer<F> {
+            const NAME: &'static str = "LtGadget";
+
+            fn configure_gadget_container(cb: &mut ConstraintBuilder<F>) -> Self {
+                let a = cb.query_cell();
+                let b = cb.query_cell();
+                let lt_gadget = LtGadget::<F, N>::construct(cb, a.expr(), b.expr());
+                cb.require_equal("a < b", lt_gadget.expr(), 1.expr());
+                LtGadgetTestContainer { lt_gadget, a, b }
+            }
+
+            fn assign_gadget_container(
+                &self,
+                input_words: &[Word],
+                region: &mut CachedRegion<'_, '_, F>,
+            ) -> Result<(), Error> {
+                let a = F::from(u64::from_le_bytes(
+                    input_words[0].to_le_bytes()[..8].try_into().unwrap(),
+                ));
+                let b = F::from(u64::from_le_bytes(
+                    input_words[1].to_le_bytes()[..8].try_into().unwrap(),
+                ));
+                let offset = 0;
+
+                self.a.assign(region, offset, Value::known(a))?;
+                self.b.assign(region, offset, Value::known(b))?;
+                self.lt_gadget.assign(region, offset, a, b)?;
+
+                Ok(())
+            }
+        }
+
+        test_math_gadget_container::<Fr, LtGadgetTestContainer<Fr>>(
+            vec![Word::from(0), Word::from(1)],
+            true,
+        );
+
+        test_math_gadget_container::<Fr, LtGadgetTestContainer<Fr>>(
+            vec![Word::from(1), Word::from((1u64 << (N * 8)) - 1)],
+            true,
+        );
+
+        test_math_gadget_container::<Fr, LtGadgetTestContainer<Fr>>(
+            vec![Word::from(0), Word::from(0)],
+            false,
+        );
+
+        // out of range check
+        test_math_gadget_container::<Fr, LtGadgetTestContainer<Fr>>(
+            vec![Word::from(1), Word::from(2 << (N * 8))],
+            false,
+        );
+    }
+}
```

Comment:
Would call this file `less_than.rs` or `lt.rs` because it's the more standard way to call this (unless there's a reason not to e.g. a conflict or something).